### PR TITLE
Replace spaces with dashes in Darwin dmg filename before GitHub upload

### PR DESCRIPTION
**Pull request by mbacchi**
Fixes https://github.com/brave/devops/issues/285

The second part of the above issue still needs fixing, so this is technically a WIP.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: script/upload.py [hunk 1]
```diff
@@ -93,8 +93,10 @@ def yield_brave_packages(dir, channel, version):
   for _, _, files in os.walk(dir):
     for file in files:
       if PLATFORM == 'darwin':
+        sought_file = 'Brave-Browser-' + channel.capitalize() + '.dmg'
         if re.match(r'Brave Browser ' + channel.capitalize() + r'.*\.dmg$', file):
-          yield file
+          os.rename(file, sought_file)
```
Comment (mihaiplesa) [inline script/upload.py]: os.rename(os.path.join(dir, file), os.path.join(dir, sought_file))

File: script/upload.py [hunk 2]
```diff
@@ -93,8 +93,10 @@ def yield_brave_packages(dir, channel, version):
...
+          yield sought_file
```
Comment (mihaiplesa) [inline script/upload.py]: elif re.match(r'Brave-Browser-' + channel.capitalize() + r'.*\.dmg$', file):
    yield sought_file

Comment (mihaiplesa) [reply script/upload.py]: this is needed in case file upload is re-attempted and the file has been previously been renamed

**Comment by bbondy** [comment]
master: 88f76cbf72015ba3204c7a5978cee654f77d8fad
0.56.x: 264f667d2e4a5baa8518f595653247c82e22e5ab
0.55.x: 793da51d4e769145d14332c4da45222f133b36b8

**Pull request merged**